lib/resources/etc_fstab.rb

```diff
@@ -0,0 +1,119 @@
+# encoding: utf-8
+# copyright:
+# author: Matthew Dromazos
+
+require 'utils/parser'
+require 'pry'
+class EtcFstab < Inspec.resource(1)
+  name 'etc_fstab'
+  desc 'Use the etc_fstab InSpec audit resource to check the configuration of the etc/fstab file.'
+  example "
+  removable_media = etc_fstab.removable_media_file_systems
+  removable_media.each do |media|
+    describe media do
+      its ( 'mount_options' ) { should include 'nosuid' }
+    end
+  end
+
+  nfs_systems = etc_fstab.nfs_file_systems
+  if removable_media != []
+    nfs_systems.each do |file_system|
+      describe file_system do
+        its ( 'mount_options' ) { should include 'nosuid' }
+        its ( 'mount_options' ) { should include 'noexec' }
+        its ( 'mount_options' ) { should include '\'sec=krb5:krb5i:krb5p\'' }
+      end
+    end
+  end
+
+  describe etc_fstab do
+    its ('home_mount_options') { should include 'nosuid' }
+  end
+  "
+
+  attr_reader :params
+
+  include CommentParser
+
+  def initialize(fstab_path = nil)
+    return skip_resource 'The `etc_fstab` resource is not supported on your OS.' unless inspec.os.linux?
+    @conf_path      = fstab_path || '/etc/fstab'
+    @files_contents = {}
+    @content        = nil
+    @params         = nil
+    read_content
+    return skip_resource '`etc_fstab` is not supported on your OS' if inspec.os.windows?
+  end
+
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:device_name,           field: 'device_name')
+        .add(:mount_point,           field: 'mount_point')
+        .add(:file_system_type,      field: 'file_system_type')
+        .add(:mount_options,         field: 'mount_options')
+        .add(:dump_options,          field: 'dump_options')
+        .add(:file_system_options,   field: 'file_system_options')
+
+  filter.connect(self, :params)
+
+  def removable_media_file_systems
+    where { !%w{xfs ext4 swap tmpfs}.include?(file_system_type) }
```

Comment:
There are a WHOLE bunch more file systems that are not removable... zfs comes to mind. And I'm pretty sure I can take a USB HDD, put ext4 on it, and it would still be removable. 

I'd like to recommend removing this method.